### PR TITLE
Fix job group update issue

### PR DESCRIPTION
**Pull request by mitiao**
See: https://progress.opensuse.org/issues/29110

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 1]
```diff
@@ -95,11 +95,16 @@ sub list {
...
 # check existing job group on top level to prevent create/update duplicate
 sub check_top_level_group {
-    my ($self) = @_;
+    my ($self, $group_id) = @_;
 
     return 0 if $self->is_parent;
     my $properties = $self->load_properties;
-    return $self->resultset->search({name => $properties->{name}, parent_id => undef});
+    unless ($group_id) {
+        return $self->resultset->search({name => $properties->{name}, parent_id => undef});
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: please don't duplicate so much code. the argument to search is a hash and it's only different in one key.

Comment (mitiao) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: fixed

File: lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm [hunk 2]
```diff
@@ -95,11 +95,18 @@ sub list {
...
 sub check_top_level_group {
-    my ($self) = @_;
+    my ($self, $group_id) = @_;
 
     return 0 if $self->is_parent;
     my $properties = $self->load_properties;
-    return $self->resultset->search({name => $properties->{name}, parent_id => undef});
+    my $conditions = {name => $properties->{name}, parent_id => undef};
+    unless ($group_id) {
+        return $self->resultset->search($conditions);
```
Comment (coolo) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: now this line is the same as 3 lines later. So please just move the $conditions->{id} = into an if - and return the search uncondiotionally

Comment (mitiao) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobGroup.pm]: yes, got it...

**Pull request merged**